src/main/java/org/folio/service/exchangerate/ExchangeRateSourceService.java

```diff
@@ -0,0 +1,92 @@
+public class ExchangeRateSourceService {
+
+  private ExchangeRateSourceDAO exchangeRateSourceDAO;
+
+  public Future<ExchangeRateSource> getExchangeRateSource(RequestContext requestContext) {
+    var dbClient = requestContext.toDBClient();
+    return dbClient.withConn(conn -> exchangeRateSourceDAO.getExchangeRateSource(conn))
+      .map(exchangeRateSource -> exchangeRateSource.orElseThrow(() ->
+        new HttpException(NOT_FOUND.getStatusCode(), NOT_FOUND.getReasonPhrase())));
+  }
+
+  public Future<ExchangeRateSource> saveExchangeRateSource(ExchangeRateSource exchangeRateSource, RequestContext requestContext) {
+    removeSensitiveData(exchangeRateSource);
+    if (!validateExchangeRateSource(exchangeRateSource)) {
+      return Future.failedFuture(new HttpException(422, ErrorCodes.EXCHANGE_RATE_SOURCE_INVALID.toError()));
+    }
+    return requestContext.toDBClient()
+      .withTrans(conn -> exchangeRateSourceDAO.saveExchangeRateSource(exchangeRateSource, conn))
+      .recover(this::handleException)
+      .onSuccess(v -> log.info("saveExchangeRateSource:: New exchange rate source was saved successfully"))
+      .onFailure(t -> log.error("Failed to save exchange rate source", t));
+  }
+
+  public Future<Void> updateExchangeRateSource(String id, ExchangeRateSource exchangeRateSource, RequestContext requestContext) {
+    removeSensitiveData(exchangeRateSource);
+    if (!validateExchangeRateSource(exchangeRateSource)) {
+      return Future.failedFuture(new HttpException(422, ErrorCodes.EXCHANGE_RATE_SOURCE_INVALID.toError()));
+    }
+    return requestContext.toDBClient()
+      .withTrans(conn -> exchangeRateSourceDAO.updateExchangeRateSource(id, exchangeRateSource, conn))
+      .recover(this::handleException)
+      .onSuccess(v -> log.info("updateExchangeRateSource:: Exchange rate source with id: '{}' was updated successfully", id))
+      .onFailure(t -> log.error("Failed to update exchange rate source with id: '{}'", id, t));
+  }
+
+  public Future<Void> deleteExchangeRateSource(String id, RequestContext requestContext) {
+    var dbClient = requestContext.toDBClient();
+    return dbClient.withTrans(conn -> exchangeRateSourceDAO.deleteExchangeRateSource(id, conn))
+      .recover(this::handleException)
+      .onSuccess(v -> log.info("deleteExchangeRateSource:: Exchange rate source with id: '{}' was deleted successfully", id))
+      .onFailure(t -> log.error("Failed to delete exchange rate source with id: '{}'", id, t));
+  }
+
+  private void removeSensitiveData(ExchangeRateSource exchangeRateSource) {
+    exchangeRateSource.setApiKey(null);
+    exchangeRateSource.setApiSecret(null);
+  }
+
+  private boolean validateExchangeRateSource(ExchangeRateSource exchangeRateSource) {
+    return exchangeRateSource != null
+      && StringUtils.isNotEmpty(exchangeRateSource.getProviderUri())
+      && StringUtils.isEmpty(exchangeRateSource.getApiKey())
+      && StringUtils.isEmpty(exchangeRateSource.getApiSecret())
```

Comment:
User can specify another apiKey and secret in UI, so we should remove isEmpty check for these 2 fields.
In future story we will integrate with some secure store and will save these 2 fields into it

Reply:
and only user with specific permission will be able to see/override apiKey secret from UI, so these keys should be configurable by admins user in UI